psi4/driver/driver_nbody.py

```diff
@@ -759,8 +776,11 @@ def assemble_nbody_components(metadata, component_results):
     results = {}
     results['nbody'] = nbody_dict
     for b in ['cp', 'nocp', 'vmfc']:
-        results['%s_energy_body_dict' % b] = eval('%s_energy_body_dict' % b)
-        results['%s_energy_body_dict' % b] = {str(i) + b: j for i, j in results['%s_energy_body_dict' % b].items()}
+        if b in metadata['bsse_type_list'] and monomer_energies != 0.0:
```

Comment:
any chance this is missing the opportunity to calculate a bsse type that has data available but wasn't explicitly requested by user?

Reply:
Okay, this one is a little tricky and might require a lot more work. Ideally, the driver would be clever enough to derive all BSSE treatments possible from the set of calculated energies, without filling blanks with zeroes. Probably too much work right now before DDD (or 1.4 release).

It should only print tables of whatever is in `'bsse_type_list'`, i.e. `return_total_data=True, bsse_type='cp'` would not print no-CP results, but would calculate them and return them on `wfn`, whereas `bsse_type=['cp','nocp']` printing and calculating both (with the order of the array changing what gets returned).

Before this change, the `2CP` field would return interaction energy for `energy(..., bsse_type='cp')` but a total energy for `energy(..., bsse_type=['cp', 'nocp'])`, which is what I want to avoid.

Reply:
Ah, got it. I think the compute-all-available would be nice down the road (and I wouldn't mind if it printed all-available, too), but no immediate need as long as it isn't a feature loss. Yes, the `2CP` you describe sounds important to avoid.